packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -521,8 +535,9 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   set offstage (bool value) {
     if (_offstage == value)
       return;
-    _offstage = value;
-    _scopeKey.currentState?._didChangeRouteOffStage();
+    setState(() {
+      _offstage = value;
```

Comment:
Sorry, I removed the comment explaining why this would cause the test failures you're seeing on Travis.  We need to set `_offstage` even if `_scopeKey.currentState` is null.  The way you've got this setup, the callback isn't run then.  Probably `setState` should run the callback unconditionally.